### PR TITLE
Bug 1336452: Required Signoffs

### PR DESCRIPTION
**Pull request by aksareen**

### CODE REVIEW

File: auslib/web/admin/swagger/api.yaml [hunk 1]
```diff
@@ -1597,74 +1963,79 @@ definitions:
...
+        maxLength: 50
+        example: releng
 
+  ChannelModel:
+    title: Channel
+    description: "model to define 'channel' field in request body."
+    type: object
+    properties:
+      channel:
+        description: The update channel of the application request an update.
```
Comment (mozbhearsum) [inline auslib/web/admin/swagger/api.yaml]: Same thing here - there's no "application requesting an update" in this context.

File: auslib/web/admin/swagger/api.yaml [hunk 2]
```diff
@@ -126,6 +132,33 @@ parameters:
...
+    description: roles are used when signing off on Scheduled Changes
+    type: string
+    minLength: 1
+    maxLength: 50
+    required: true
+
+  channelRequiredParam:
+    name: channel
+    in: query
+    description: The update channel of the application request an update.
```
Comment (aksareen) [inline auslib/web/admin/swagger/api.yaml]: I borrowed the definition from the [database model in docs](http://mozilla-balrog.readthedocs.io/en/latest/database.html). What should I change it to ?

File: auslib/web/admin/swagger/api.yaml [hunk 3]
```diff
@@ -841,7 +874,36 @@ paths:
                 type: array
                 minItems: 0
                 items:
-                  $ref: '#/definitions/ReleaseRevision'
+                  type: object
```
Comment (aksareen) [inline auslib/web/admin/swagger/api.yaml]: Previously I had defined 'ReleaseRevision' model to define the object as element for every item in revisions array above. However, I realized that the ReleaseRevision model won't re-used at all in any other API, hence I moved it out from definitions.

File: auslib/web/admin/swagger/api.yaml [hunk 4]
```diff
@@ -1597,74 +1963,79 @@ definitions:
 
   DataVersionModel:
     title: Data Version
-    description: version number of object
+    description: "model to re-use 'data_version' field."
```
Comment (aksareen) [inline auslib/web/admin/swagger/api.yaml]: How about concat of both "model to re-use 'data_version' field. Data_version is the version  number of object" ?

File: auslib/web/admin/swagger/api.yaml [hunk 5]
```diff
@@ -1597,74 +1963,79 @@ definitions:
...
-  ReleaseBase:
-    title: Release Base
-    description: "To Balrog, a 'release' is data about a related set of builds. This does _not_ match up with the concept of a 'release' being on the 'beta', 'release' or 'esr' channel elsewhere. In Balrog, each set of nightlies on any branch is considered a release"
+  SignoffsRequiredModel:
+    title: Signoffs Required
+    description: "model to re-use 'signoffs_required' field."
+    type: object
+    properties:
+      signoffs_required:
+        description: minimum number of signoffs required to execute the change
```
Comment (aksareen) [inline auslib/web/admin/swagger/api.yaml]: this model only defines the signoffs_required integer field defined in the [forms](https://github.com/mozilla/balrog/blob/eb35f9938d327719edc2fb330edc9476378caf41/auslib/web/admin/views/forms.py#L381) . Am not entirely getting your point above.

File: auslib/web/admin/swagger/api.yaml (hunk 1 above)
Comment (aksareen) [inline auslib/web/admin/swagger/api.yaml]: that's the same definition of channel in the docs as above. What should I change this to ?

File: auslib/web/admin/swagger/api.yaml (hunk 3 above)
Comment (mozbhearsum) [inline auslib/web/admin/swagger/api.yaml]: OK. In the future, this sort of thing would be better as its own PR, as the scope here was supposed to be just required signoffs.

File: auslib/web/admin/swagger/api.yaml (hunk 4 above)
Comment (mozbhearsum) [inline auslib/web/admin/swagger/api.yaml]: Good enough for now.

**Comment by aksareen** [comment]
@mozbhearsum : The [UI issue](https://screenshots.firefoxusercontent.com/images/c10bfdd4-19e9-4299-ac20-7f93ba42eb4b.png) seems to be similar to one we encountered during Rules API migration. it seems that the input json of signoffs is passed as `{"signoffs_required": u'1'}` i.e. a string. Seems a custom validator needs to be added to verify the value as done previously for Rule model's fields

File: auslib/web/admin/swagger/api.yaml (hunk 5 above)
Comment (mozbhearsum) [inline auslib/web/admin/swagger/api.yaml]: What I'm trying to say is that the signoffs_required is linked with a role, and there may be multiple combinations of role+signoffs_required *before* a change can be executed, so having "to execute the change" here (as part of a single role) is a bit confusing. Perhaps something like "The number of signoffs required for a given role" would be better?

File: auslib/web/admin/swagger/api.yaml (hunk 2 above)
Comment (mozbhearsum) [inline auslib/web/admin/swagger/api.yaml]: Alright, I guess that's fine for now.

File: auslib/web/admin/swagger/api.yaml (hunk 3 above)
Comment (aksareen) [inline auslib/web/admin/swagger/api.yaml]: will do

**Pull request merged**